internal/backend/remote-state/http/client_test.go

```diff
@@ -251,3 +255,136 @@ func TestHttpClient_IsLockingEnabled(t *testing.T) {
 		})
 	}
 }
+
+// Tests the Lock method for the HTTP client.
+// Test to see correct lock info is returned
+func TestHttpClient_lock(t *testing.T) {
+	stateLockInfoA := statemgr.LockInfo{
+		ID:        "ada-lovelace-state-lock-id",
+		Who:       "AdaLovelace",
+		Operation: "TestTypePlan",
+	}
+
+	trimString := func(str string) string {
+		// Anonymous Helper function to remove new line, tab
+		// and space characters from a string
+		space := regexp.MustCompile(`\s+`)
+		return space.ReplaceAllString(str, " ")
+	}
+
+	testCases := []struct {
+		name           string
+		lockMethod     string
+		lockInfo       *statemgr.LockInfo
+		handler        http.HandlerFunc
+		validateResult func(lockID string, errorMessage error)
+	}{
+		{
+			// Successful locking HTTP remote state
+			name:       "Successfully locked",
+			lockMethod: "LOCK",
+			lockInfo:   &stateLockInfoA,
+			handler: func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusOK)
+			},
+			validateResult: func(lockID string, errorMessage error) {
+				expectedStateLockID := stateLockInfoA.ID
+				if lockID != stateLockInfoA.ID {
+					t.Errorf("Lock, lockID = %q, want %q", lockID, expectedStateLockID)
+				}
+				if errorMessage != nil {
+					t.Errorf("Lock, error message is not nil %v", errorMessage)
+				}
+			},
+		},
+		{
+			// Failed to lock state, HTTP remote state already locked
+			name:       "Locked remote state",
+			lockMethod: "LOCK",
+			lockInfo:   &stateLockInfoA,
+			handler: func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusLocked)
+				w.Write([]byte(`{"ID":"linus-torvalds-http-remote-state-lock-id", "Path": "", "Operation": "TestTypePlan", "Who": "LinusTorvalds" }`))
+			},
+			validateResult: func(lockID string, errorMessage error) {
+				if lockID != "" {
+					t.Errorf("Lock, lockID should be an empty string, got %q", lockID)
+				}
+				if errorMessage == nil {
+					t.Errorf("Lock, expected an error when trying to lock a state that is already locked. got %q", errorMessage)
+				}
+
+				errString := trimString(errorMessage.Error())
+				expectedStateLockID := "linus-torvalds-http-remote-state-lock-id"
+
+				count := strings.Count(errString, expectedStateLockID)
+				if count != 2 {
+					t.Fatalf("Lock, expected lock id %q to occur 2 times, got %v", expectedStateLockID, count)
+				}
+
+				if !strings.Contains(errString, fmt.Sprintf("HTTP remote state already locked: ID=%s", expectedStateLockID)) {
+					t.Fatalf("Lock, expected locked: ID= to be %q, got %q", expectedStateLockID, errorMessage.Error())
+				}
+
+				if !strings.Contains(errString, fmt.Sprintf("ID: %s", expectedStateLockID)) {
+					t.Fatalf("Lock, expected ID: to be %q, got %q", expectedStateLockID, errorMessage.Error())
+				}
+
+				if !strings.Contains(errString, "Who: LinusTorvalds") {
+					t.Fatalf("Lock, expected Who: to be LinusTorvalds, got %q", errorMessage.Error())
+				}
```

Comment:
Can we compare two strings (error messages) directly?

PS with my previous suggestion, this comparison can be moved to a shared test run logic (inside a loop).

Reply:
I have updated the comparison checks